[Workers] page rules and workers flow (#3212)
cloudflare/cloudflare-docs: products/workers/src/content/platform/workers-with-page-rules.md

---
order: 
pcx-content-type: concept
---

# Page Rules

Page Rules trigger certain actions whenever a request matches one of the URL patterns you define. You can define a page rule to trigger one or more actions whenever a certain URL pattern is matched. Refer to the [Page Rules Support article](https://support.cloudflare.com/hc/en-us/articles/218411427) to learn more about configuring Page Rules.

## Page Rules with Workers

> Cloudflare acts as a [reverse proxy](https://www.cloudflare.com/learning/what-is-cloudflare/) to provide services, like Page Rules, to Internet properties. Traffic headed to your application will first visit the closest Cloudflare data center to the visitor, where services like Workers and Page Rules will run, before going, if needed, to your origin server. If your application is built on Workers and/or Pages, the [Cloudflare network edge](https://www.cloudflare.com/learning/serverless/glossary/what-is-edge-computing/) acts as your origin server and responds to requests directly from the Cloudflare edge. 

Reviewer: I don't think we need to keep the "your origin server" part because whether or not something falls thru to their hardware is entirely dependent on what Page Rules are used, what their Workers are doing, etc. Not even sure we need the last `If your application is built...` sentence – probably not the appropriate place to tell (or re-tell/summarize) origin servers

```suggestion
Cloudflare acts as a [reverse proxy](https://www.cloudflare.com/learning/what-is-cloudflare/) to provide services, like Page Rules, to Internet properties. Your application's traffic will pass through a Cloudflare data center that is closest to the visitor. There are hundreds of these around the world, each of which are capable of running services like Workers and Page Rules. If your application is built on Workers and/or Pages, the [Cloudflare network edge](https://www.cloudflare.com/learning/serverless/glossary/what-is-edge-computing/) acts as your origin server and responds to requests directly from the Cloudflare edge. 
```

When using Page Rules with Workers, the following workflow is applied. 

> 1. Request arrives at Cloudflare data center.

Reviewer 2: nit: I would maybe add articles here and on step 5.? like `A request`, or `The Worker`

> 1. Request arrives at Cloudflare data center.

Author: ```suggestion
1. A request arrives at Cloudflare data center.
```

2. Cloudflare decides if this request is a Worker route.

> 3. Because this is a Worker route, Cloudflare disables a number of features, including some that would be set by Page Rules.

Reviewer 3: "a number of features" is pretty vague. As someone programming on this platform, I need to know more details. For example, does `Forwarding URL` apply before or after a worker? That different is _huge_.

4. Page Rules run as part of normal request proessing with some features now disabled.
5. Worker executes.

> If you are experiencing Page Rule errors when running Workers, contact your Cloudflare account team.

Author: ```suggestion
If you are experiencing Page Rule errors when running Workers, contact your Cloudflare account team or [Cloudflare Support](https://support.cloudflare.com/hc/en-us/articles/200172476-Contacting-Cloudflare-Support).
```
